main/src/ca/uwaterloo/flix/language/ast/Kind.scala

```diff
@@ -39,6 +39,25 @@ sealed trait Kind {
     */
   override def toString: String = Kind.ShowInstance.show(this)
 
+  /**
+    * Returns true if `left` is a subkind of `this`.
+    * Right-associative.
+    */
+  def <::(left: Kind): Boolean = (left, this) match {
+    // subkinds
+    case (Kind.Record, Kind.Star) => true
+    case (Kind.Schema, Kind.Star) => true
+
+    // identities
+    case (Kind.Star, Kind.Star) => true
+    case (Kind.Bool, Kind.Bool) => true
+    case (Kind.Record, Kind.Record) => true
+    case (Kind.Schema, Kind.Schema) => true
+
+    case (Kind.Arrow(k11, k12), Kind.Arrow(k21, k22)) => (k11 <:: k21) && (k12 <:: k22)
+    case _ => false
+  }
```

Comment:
Much better; but can I ask you to move the identities first?

I think this not only makes the function easier to understand, it is also likely more performant since they are more common.

Reply:
Yep, makes sense.